ibm/data_source_ibm_dl_provider_gateways_test.go

```diff
@@ -0,0 +1,32 @@
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccIBMDLProviderGWsDataSource_basic(t *testing.T) {
+	name := "dl_provider_gws"
+	resName := "data.ibm_dl_provider_gateways.test_dl_provider_gws"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMDLProviderGWsDataSourceConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resName, "gateways.0.id"),
```

Comment:
Change this condition if you don't have any gateways in your account it gives an error gateways.0.id